### PR TITLE
Add a note of known limitations in code processing

### PR DESCRIPTION
**Pull request by rakhimov**
This note documents some limitations of Lizard
and provides some contract for developers
to keep Lizard's code simpler.

### CODE REVIEW

**Comment by terryyin** [comment]
I think Lizard should aim at "no crashes and no dead loops" for incorrect & incomplete code. But doesn't need to provide warning or error message for incorrect & incomplete code.
And I should try to achieve "nearly linear correlation" to the completeness of the code.
So that potentially lizard can work at the background of an editor in real time.

What do you think?

**Comment by rakhimov** [comment]
That's an interesting goal I didn't consider. Let me think.

**Comment by rakhimov** [comment]
I think I went to far requiring the behavior to be undefined.
In case of real time processing, I think your idea of soft guarantees on best effort is better.
I should also note that "undefined behavior" and Python tool doesn't sound good.

File: README.rst [hunk 1]
```diff
@@ -265,6 +265,29 @@ before a function it will suppress the warning for that function.
...
    }
 
+
+Limitations
+-----------
+
+Lizard requires syntactically correct code.
+Upon processing input with incorrect or unknown syntax,
+Lizard's behavior is pretty much undefined (in good old C sense).
+It can ignore the code, throw an exception, crash, fail silently, or succeed.
```
Comment (rakhimov) [inline README.rst]: I am trying to think about how to reword this. The following is just brain dump.
```
In the case of invalid/partial/incorrect/unknown syntax:

- Guarantees (or tries its best to) no hard failures: exit, crash, exceptions.
- There is a chance of a combination of the following soft failures:
    - omission
    - misinterpretation
    - improper analysis / tally
    - success (the code in consideration is not relevant, e.g., global macros in C)

Developers of Lizard attempt to minimize the possibility of soft failures.
Hard failures are bugs in Lizard code,
while soft failures are trade-offs.
```

Comment (rakhimov) [inline README.rst]: This means that the behavior is defined.

Comment (rakhimov) [inline README.rst]: + (As @terryyin  noted) it guarantees to terminate.

**Pull request merged**